spec/requests/blocked_users_n_plus_one_spec.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+
+describe "n+1 queries in the blocked users controller" do
+  include LoginMacros
+
+  describe "#index" do
+    context "with a logged in user who has blocked someone", n_plus_one: true do
+      let!(:blocker) { create(:user) }
+
+      populate do |n|
+        blocked_users = create_list(:user, n)
+        blocked_users.each do |blocked|
+          Block.create(blocker: blocker, blocked: blocked)
+        end
+      end
+
+      before do
+        fake_login_known_user(blocker)
+      end
+
+      subject do
+        proc do
+          get user_blocked_users_path(user_id: blocker)
+        end
+      end
+
+      warmup { subject.call }
+
+      it "produces a constant number of queries" do
+        expect { subject.call }
+          .to perform_constant_number_of_queries
```

Comment:
This needs the `with_attached_*` here: https://github.com/otwcode/otwarchive/blob/1588f6c552b1c85adeb1b30e31411cb2cb6f37d7/app/controllers/blocked/users_controller.rb#L14